src/main/java/uk/gov/companieshouse/documentsigningapi/coversheet/CoverSheetService.java

```diff
@@ -1,39 +1,188 @@
+
+        insertText(contentStream, PAGE_HEADING, PDType1Font.HELVETICA_BOLD, 30, 770);
+        insertText(contentStream, getTodaysDate(), PDType1Font.HELVETICA, 18, 750);
+        insertText(contentStream, CERTIFIED_DOCUMENT_TYPE, PDType1Font.HELVETICA_BOLD, 24, 650);
+
+        textWrapper(contentStream, getCompany(coverSheetData), 18, DEFAULT_LEFT_MARGIN, 620);
+        textWrapper(contentStream, getFilingHistory(coverSheetData), 18, DEFAULT_LEFT_MARGIN, 590);
+        textWrapper(contentStream, DOCUMENT_SIGNED_TEXT, 18, DEFAULT_LEFT_MARGIN, 560);
+
+        insertText(contentStream, PAGE_SPACER, PDType1Font.HELVETICA, 18, 530);
+        insertText(contentStream, VIEW_FILE_HEADING, PDType1Font.HELVETICA_BOLD, 18, 480);
+
+        contentStream.drawImage(signatureImage, DEFAULT_LEFT_MARGIN, 420, INFORMATION_SECTION_IMAGE_WIDTH, INFORMATION_SECTION_IMAGE_HEIGHT);
+        textWrapper(contentStream, SIGNATURE_HELPTEXT_LINE_1, 14, OFFSET_TO_RIGHT_OF_IMAGES, 440);
+
+        renderTextWithLink(
+                SIGNATURE_HELPTEXT_LINE_2_START,
+                SIGNATURE_HELPTEXT_LINE_2_END,
+                new Link(SIGNATURE_HELPTEXT_LINE_2_LINK, ADOBE_DOWNLOAD_URL),
+                new Font(PDType1Font.HELVETICA, 14),
+                coverSheet,
+                contentStream,
+                new Position(OFFSET_TO_RIGHT_OF_IMAGES, 420));
+
+        contentStream.drawImage(emailImage, DEFAULT_LEFT_MARGIN, 370, INFORMATION_SECTION_IMAGE_WIDTH, INFORMATION_SECTION_IMAGE_HEIGHT);
+        textWrapper(contentStream, EMAIL_HELPTEXT, 14, OFFSET_TO_RIGHT_OF_IMAGES, 385);
+
+        contentStream.drawImage(printerImage, DEFAULT_LEFT_MARGIN, 325, INFORMATION_SECTION_IMAGE_WIDTH, INFORMATION_SECTION_IMAGE_HEIGHT);
+        textWrapper(contentStream, PRINTER_HELPTEXT, 14, OFFSET_TO_RIGHT_OF_IMAGES, 340);
+
+        contentStream.close();
+    }
+
+    private void insertText(PDPageContentStream contentStream, String text, PDType1Font font,
+                            float fontSize, float yPosition) throws IOException {
+        contentStream.beginText();
+        contentStream.setFont(font, fontSize);
+        contentStream.newLineAtOffset(DEFAULT_LEFT_MARGIN, yPosition);
+        contentStream.showText(text);
+        contentStream.endText();
+    }
+
+    private void textWrapper(PDPageContentStream contentStream, String textToWrap, float fontSize, float xPosition, float yPosition) throws IOException {
+        String[] wrappedText = WordUtils.wrap(textToWrap, 80).split("\\r?\\n");
+
+        for(int i = 0; i < wrappedText.length; i++){
+            contentStream.beginText();
+            contentStream.setFont(PDType1Font.HELVETICA, fontSize);
+            contentStream.newLineAtOffset(xPosition,yPosition-i*15);
+            contentStream.showText(wrappedText[i]);
+            contentStream.newLine();
+            contentStream.endText();
+        }
     }
```

Comment:
Does this section handle the insanely long maximum company name as covered in the PoC?

Reply:
I don't know. @caismith-ch What do you think?

Reply:
(From Hywel's POC).